### ISSUE TITLE
Resource not accessible by integration when using GitHub App

### ISSUE THREAD

**Issue by mike-pisman** [peter-evans/repository-dispatch#270]
Hi,

I have a problem when using the action or manually sending HTTP request, I get `Resource not accessible by integration`

One difference from similar issues is that I am using a GitHub App. I have followed [this guide](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/making-authenticated-api-requests-with-a-github-app-in-a-github-actions-workflow) to get access token from the app.

---

<details>
  <summary>My workflow</summary>

```yaml
jobs:
  docs: 
    runs-on: arc-runner-k8s
    permissions:
      actions: write
      contents: write
    steps:
      - name: Generate a token
        id: generate_token
        uses: actions/create-github-app-token@v1
        with:
          app_id: ${{ secrets.APP_ID }}
          private_key: ${{ secrets.APP_PEM }}
      - name: Repository Dispatch
        uses: peter-evans/repository-dispatch@v2
        with:
          token: ${{ steps.generate_token.outputs.token }}
          repository: unipoll/gollum-test
          event-type: update_submodules
```

</details>

---

`arc-runner-k8s` is the self hosted action runner controller, which is authenticated with the same app. 
According to [this article](https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28), I have set app permissions to allow `repository dispatch requests`. In repository permissions: `contents: read and write`, `actions: read and write` , `metadata: read only`

---

<details>
  <summary>Permission screenshot</summary>
  
  Spoiler text. Note that it's important to have a space after the summary tag. You should be able to write any markdown you want inside the `<details>` tag... just make sure you close `<details>` afterward.
  
![image](https://github.com/peter-evans/repository-dispatch/assets/58354623/d3401f1c-9b88-4ecd-b34a-cd2d76fe27c4)

</details>

<details>
  <summary>App settings page</summary>

![image](https://github.com/peter-evans/repository-dispatch/assets/58354623/24dd1c23-0689-4475-bc11-11f1965f2578)

</details>

I was hoping someone might have experience with the GitHub apps and/or this issue. Thanks

**Comment by mike-pisman** [comment]
Hi @peter-evans, I am currently using `workflow_dispatch` on the main branch of non-forked repository.
The full workflow:
```yaml
name: Update docs

on:
  workflow_dispatch:
  gollum:

jobs:
  docs: 
    runs-on: arc-runner-k8s
    permissions:
      actions: write
      contents: write
    steps:
      - name: Generate a token
        id: generate_token
        uses: actions/create-github-app-token@v1
        with:
          app_id: ${{ secrets.APP_ID }}
          private_key: ${{ secrets.APP_PEM }}
      - name: Repository Dispatch
        uses: peter-evans/repository-dispatch@v2
        with:
          token: ${{ steps.generate_token.outputs.token }}
          repository: unipoll/gollum-test
          event-type: update_submodules
```

And here is a link to the job: https://github.com/unipoll/angular/actions/runs/6387166198

**Comment by mike-pisman** [comment]
I've tried generating JWT token, sending HTTP requests with curl, using `--user "APP_ID:APP_SECRETE` instead of token, but nothing seems to work when trying to perform request with app credentials, I still get `Resource not accessible by integration`. I must be doing something wrong with app configuration, even though I have set all the permissions as recommended in documentation.

At last I tried using PAT token, just to make sure it wasn't some other issue. With the PAT token, the workflow and `repository-dispatch` action worked and successfully triggered a workflow in the `gollum-test` repository. 

Using PAT, however, does not work for me, so I will stick to manually triggering the workflow for now. 

I would appreciate, if you keep this issue open, in case someone else, who's also using GitHub App, will figure it out the solution. If I manage to resolve the problem, I will make sure to post it here.
